emsdk.py

```diff
@@ -2900,6 +2936,14 @@ def extract_bool_arg(name):
       return True
     return False
 
+  def extract_string_arg(name):
+    nonlocal args
+    for i in range(len(args)):
+      if args[i] == name:
+        value = args[i + 1]
+        args = args[:i] + args[i + 2:]
```

Comment:
Rather then re-assigning `args` you can maybe avoid the `nonlocal` thing by doing `del args[i:i+1]` which is in-place.